src/consumer.ts

```diff
@@ -142,7 +143,36 @@ export class Consumer extends TypedEventEmitter {
       this.emit('aborted');
     }
 
-    this.emit('stopped');
+    if (options?.waitForInFlightMessagesMs > 0) {
+      this.waitForInFlightMessagesToComplete(
+        options.waitForInFlightMessagesMs
+      ).then(() => {
+        this.emit('stopped');
+      });
+    } else {
+      this.emit('stopped');
+    }
+  }
+
+  /**
+   * Wait for in flight messages to complete.
+   * @param {number} waitTimeoutMs
+   * @private
+   */
+  private async waitForInFlightMessagesToComplete(
+    waitTimeoutMs: number
+  ): Promise<void> {
+    const startedAt = Date.now();
+    while (this.isPolling) {
```

Comment:
I wonder if this is better as a setTimeout that checks if the consumer is polling, and re triggers after a few ms if it isn't.